### PR TITLE
Add KeyDataReader and KeyDataWriter implementations for LUKS

### PR DESCRIPTION
**Pull request by chrisccoulson**
This adds support for reading and writing KeyData objects to the LUKS
header. A follow up PR will add support for making use of these during
activation.

### CODE REVIEW

File: keydata_luks.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+// Priority indicates the priority of the keyslot associated with the token from
+// which this KeyData is read. The default priority is 0 with higher numbers
+// indicating a higher priority.
+func (r *LUKS2KeyDataReader) Priority() int {
+	return r.priority
+}
+
+// NewLUKS2KeyDataReader is used to read a LUKS2 token containing key data with
+// the specified name on the specified LUKS2 container.
+func NewLUKS2KeyDataReader(devicePath, name string) (*LUKS2KeyDataReader, error) {
```
Comment (pedronis) [inline keydata_luks.go:59]: I never remarked this but golang style usually put constructs/New functions first after the struct definition before the methods. secboot doesn't seem to follow this convention. Might be easier to do a separate PR to address this at some point

File: keydata_luks.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+// NewLUKS2KeyDataWriter creates a new LUKS2KeyDataWriter for atomically writing a
+// KeyData to a LUKS2 token with the specicied name and priority on the specified
+// LUKS2 container.
+//
+// The container must already contain a token of the correct type with the supplied
+// name. The initial token is bootstrapped by InitializeLUKS2Container or
+// SetLUKS2ContainerUnlockKey.
+func NewLUKS2KeyDataWriter(devicePath, name string, priority int) (*LUKS2KeyDataWriter, error) {
```
Comment (pedronis) [inline keydata_luks.go:114]: shouldn't there be a way to reuse the priority of the preexisting token/slot ? without further context is a bit unclear why we would re-specify this all the times

File: keydata_luks.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	token, _, exists := view.TokenByName(name)
+	if !exists {
+		return nil, errors.New("a keyslot with the specified name does not exist")
+	}
+
+	kdToken, ok := token.(*luksview.KeyDataToken)
+	if !ok {
+		return nil, errors.New("named keyslot is the wrong type")
```
Comment (pedronis) [inline keydata_luks.go:72]: s/is/has/ ? as below

**Review by pedronis — COMMENTED** [review]
did a pass, some comments/questions

File: keydata_luks.go (hunk 1 above)
Comment (chrisccoulson) [reply keydata_luks.go:42]: Thanks, I've reordered this one.

File: keydata_luks.go (hunk 2 above)
Comment (chrisccoulson) [reply keydata_luks.go]: I've removed the priority argument from this function and added a SetPriority function instead.

**Review by pedronis — APPROVED** [review]
thank you
